### ISSUE TITLE
Trying to do printf-style formatted output with format and failing

### ISSUE THREAD

**Issue by donaldcallen** [ziglang/zig#4823]
Using zig-linux-x86_64-0.5.0+39589cffe.tar.xz on a Slackware Linux system (current; up-to-date):

I encounter this error:

zig build-exe --main-pkg-path .. --single-threaded -I/usr/include -L/usr/local/lib64 -L/usr/lib64 -lsqlite3 -lc `pkg-config --libs gtk+-3.0` composite_register.zig
Semantic Analysis [701/1185] /usr/local/lib/zig/std/fmt.zig:319:22: error: container 'std.fs.file.File' has no member called 'Error'
) @TypeOf(out_stream).Error!void {
                     ^
/usr/local/lib/zig/std/fmt.zig:319:28: note: called from here
) @TypeOf(out_stream).Error!void {
                           ^
./composite_register.zig:86:19: note: called from here
        fmt.format(stdout, "{}" ++ sep ++ "{}" ++ sep ++ "{}\n",
                  ^
./composite_register.zig:26:20: note: called from here
pub fn main() void {

**Comment by haze** [comment]
I think you want `std.debug.warn`

**Comment by donaldcallen** [comment]
Then what is format for?

**Comment by fengb** [comment]
stdout is a File. You need to convert it into an outStream with `stdout.outStream()`

**Comment by donaldcallen** [comment]
Yes, I see that now that you point it out. But this is not a useful way  to inform me of a type mis-match, so I think this still needs some attention.

Thanks --

**Comment by slimsag** [comment]
Just posting to say I ran into this as well, in a different context, and was confused.

![image](https://user-images.githubusercontent.com/3173176/107870151-789f9f80-6e53-11eb-99b7-ad24af9edb2c.png)
